system-x/services/db/cassandra/src/main/java/software/tnb/db/cassandra/resource/openshift/OpenshiftCassandra.java

```diff
@@ -157,16 +158,22 @@ public void closeResources() {
 
     @Override
     public boolean isReady() {
-        List<Pod> pods = OpenshiftClient.get().getLabeledPods(OpenshiftConfiguration.openshiftDeploymentLabel(), name());
-        if (ResourceFunctions.areExactlyNPodsReady(1).apply(pods)) {
-            return OpenshiftClient.get().getLogs(pods.get(0)).contains("Startup complete");
+        final PodResource<Pod> pod = servicePod();
+        if (pod.isReady()) {
+            return OpenshiftClient.get().getLogs(pod.get()).contains("Startup complete");
         }
         return false;
     }
 
     @Override
     public boolean isDeployed() {
-        return OpenshiftClient.get().getLabeledPods(OpenshiftConfiguration.openshiftDeploymentLabel(), name()).size() > 0;
+        return OpenshiftClient.get().apps().deployments().withLabel(OpenshiftConfiguration.openshiftDeploymentLabel(), name()).list()
+            .getItems().size() > 0;
+    }
+
+    @Override
+    public Predicate<Pod> podSelector() {
+        return p -> OpenshiftClient.get().hasLabels(p, Map.of(OpenshiftConfiguration.openshiftDeploymentLabel(), name()));
```

Comment:
This will be the same for quite a lot of services but I'm not sure if making podSelector default to this is also the right solution. Maybe we could put the predicate into a const field that we could use for this "default" predicate?